hack/conformance.sh

```diff
@@ -12,11 +12,25 @@ fi
 export KUBECONFIG=$(cluster-up/kubeconfig.sh)
 
 teardown() {
+    rv=$?
     ./sonobuoy status --json
     ./sonobuoy logs > ${ARTIFACTS}/sonobuoy.log
     results_tarball=$(./sonobuoy retrieve)
     tar -xvzf $results_tarball plugins/e2e/results/
     cp -f $(find plugins/e2e/results/* -name "*.xml") ${ARTIFACTS}/
+
+    if [ $rv -ne 0 ]; then
+        echo "error found, exiting"
+        exit $rv
```

Comment:
Are we sure we need this ? I expect the exit not being calculated by the teardown function called by trap.

Reply:
this is if the main script failed before the teardown and we dont want to mask this error

Reply:
understood

Reply:
Just to understand, Is there special reason to use `exit` and not `return` inside `teardown()` function?